src/codeu/chat/server/Controller.java

```diff
@@ -115,6 +122,10 @@ public User newUser(Uuid id, String name, Time creationTime) {
           name,
           creationTime);
 
+      if(!retrieveOn){
```

Comment:
i was thinking of maybe modifying the newUser method, and those similar to it, to include a parameter whether to log stuff.